include/zephyr/kernel/mm.h

```diff
@@ -114,6 +114,25 @@ extern "C" {
  */
 #define K_MEM_MAP_LOCK		BIT(17)
 
+/**
+ * Region will be unpaged i.e. not mapped into memory
+ *
+ * This is meant to be used by kernel code and not by application code.
+ *
+ * Corresponding memory address range will be set so no actual memory will
+ * be allocated initially. Allocation will happen through demand paging when
+ * addresses in that range are accessed. This is incompatible with
+ * K_MEM_MAP_LOCK.
+ *
+ * When this flag is specified, the phys argument to k_mem_map_phys_guard()
+ * is interpreted as a backing store location value not a physical address.
+ * This is very similar to arch_mem_page_out() in that regard.
```

Comment:
This is going to be confusing to users of `k_mem_map_phys_guard()` because now the argument may mean something totally different depending on flags, and will definitely prone to coding error. `arch_mem_page_out()` is considered an internal API so this is somehow okay as the users of this should be confined inside the kernel.